packages/services/src/serverconnection.ts

```diff
@@ -93,14 +93,14 @@ namespace ServerConnection {
     dataType?: string;
 
     /**
-     * The outgoing content type, used to set the `Content-Type` header.  Defaults to `'application/json'` when there is sent data.
+     * The outgoing content type, used to set the `Content-Type` header.
      */
     contentType?: string;
 
     /**
      * The request data.
      */
-    data?: any;
+    data?: Blob | BufferSource | FormData | URLSearchParams | ReadableStream | string;
```

Comment:
We ensure there is always data on the methods that should have a payload (otherwise it can be perceived by the server as an attack vector).  If there is no data we inject '{}' and give it a `Content-Type`.